### PR TITLE
📖 Fix typo in docs/proposals

### PR DESCRIPTION
**Pull request by chymy**
**What this PR does / why we need it**:
Fix typo in docs/proposals
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/proposals/20181121-machine-api.md [hunk 1]
```diff
@@ -42,9 +42,9 @@ performed in-place or via Node replacement.
 ## Proposal
 
 This proposal introduces a new API type: Machine. See the full definition in
-[types.go](types.go).
+[machine_types.go](https://github.com/kubernetes-sigs/cluster-api/tree/release-0.2/pkg/apis/deprecated/v1alpha1/machine_types.go).
```
Comment (sbueringer) [inline docs/proposals/20181121-machine-api.md:45]: Maybe we shouldn't link to release-0.2

**Comment by sbueringer** [comment]
one nit otherwise lgtm
/ok-to-test

File: docs/proposals/20181121-machine-api.md (hunk 1 above)
Comment (chymy) [reply docs/proposals/20181121-machine-api.md:45]: Thanks, the proposal is old, should we use the old API or master branch?

Comment (chymy) [reply docs/proposals/20181121-machine-api.md:45]: The hyperlinks should also be modified in
https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/proposals/20181121-machine-api.md#types

Comment (sbueringer) [reply docs/proposals/20181121-machine-api.md]: Let's wait for feedback from other folks. I'm not sure what the policy is.

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**